src/main.cpp

```diff
@@ -137,9 +137,11 @@ void report_odrv(const std::string_view& symbol, const die& a, const die& b, dw:
     // this is that (as this comment is written) we don't detect supersets.
     // So if C has more methods than A and B it may not be detected.
     static std::set<std::size_t> unique_odrv_types;
-    std::size_t symbol_hash = hash_combine(0, symbol, odrv_category);
-    bool did_insert = unique_odrv_types.insert(symbol_hash).second;
-    if (!did_insert) return; // We have already reported an instance of this.
+    if (settings._filter_redundant) {
+        std::size_t symbol_hash = hash_combine(0, symbol, odrv_category);
+        bool did_insert = unique_odrv_types.insert(symbol_hash).second;
+        if (!did_insert) return; // We have already reported an instance of this.
+    }
```

Comment:
If you tuck the `std::set` into the if block, it'll never get initialized with `_filter_redundant` off, saving you a little in time and space.
```suggestion
    if (settings._filter_redundant) {
        static std::set<std::size_t> unique_odrv_types;
        std::size_t symbol_hash = hash_combine(0, symbol, odrv_category);
        bool did_insert = unique_odrv_types.insert(symbol_hash).second;
        if (!did_insert) return; // We have already reported an instance of this.
    }
```